pkg/codegen/dotnet/gen.go

```diff
@@ -2076,7 +2076,7 @@ func genProjectFile(pkg *schema.Package,
 		// only add a package reference to Pulumi if we're not referencing a local Pulumi project
 		// which we usually do when testing schemas locally
 		if !referencedLocalPulumiProject {
-			packageReferences["Pulumi"] = "[3.23.0,4)"
+			packageReferences["Pulumi"] = "[3.51.0,4)"
```

Comment:
What does this actually do?  Are we checking for the currently released version?

Reply:
This tells the SDK to use at least Pulumi.nuget 3.51 (which targets net6). I'm not sure it's strictly needed, I think a net6 library can reference a net3.1 library, I can check that quickly.

Reply:
Yeh actually this seems to work, so we could probably drop this.